pennylane/_qubit_device.py

```diff
@@ -1000,8 +1027,14 @@ def sample(self, observable, shot_range=None, bin_size=None):
                 ) from e
 
         if bin_size is None:
+            if counts:
+                return self._samples_to_counts(samples, no_observable_provided)
             return samples
-
+        if counts:
+            return [
```

Comment:
Is this line being covered? Could likely be done by defining a shot vector on the device.

Reply:
Thanks! No worries there.